From 35f893f1c3a62bc7d97669b1449f6f10c81e316e Mon Sep 17 00:00:00 2001
From: Thecrazyskull <anaskarbila@aospa.co>
Date: Fri, 23 Dec 2016 18:08:03 -0600
Subject: [PATCH 124/172] ListPreference: prevent
 ArrayIndexOutOfBoundsException

* Index value is not allowed to be higher/equal to the array length

Change-Id: Iff361603f337e4e77085717fce6227815ea06fc7
---
 src/com/android/camera/ListPreference.java | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/com/android/camera/ListPreference.java b/src/com/android/camera/ListPreference.java
index 05ab10f07..50afc0849 100644
--- a/src/com/android/camera/ListPreference.java
+++ b/src/com/android/camera/ListPreference.java
@@ -207,6 +207,7 @@ public class ListPreference extends CameraPreference {
         ArrayList<CharSequence> entries = new ArrayList<CharSequence>();
         ArrayList<CharSequence> entryValues = new ArrayList<CharSequence>();
         for (int i = 0, len = mEntryValues.length; i < len; i++) {
+            if (i >= mEntries.length) break;
             if (supported.indexOf(mEntryValues[i].toString()) >= 0) {
                 entries.add(mEntries[i]);
                 entryValues.add(mEntryValues[i]);
-- 
2.17.1

